runtime-modules/content/src/benchmarks/mod.rs

```diff
@@ -0,0 +1,541 @@
+    voucher_objs_number_limit: u64,
+) where
+    T: Config + storage::Config,
+{
+    let storage_wg_leader_signed = RawOrigin::Signed(lead_account_id);
+    Storage::<T>::update_storage_buckets_voucher_max_limits(
+        T::Origin::from(storage_wg_leader_signed),
+        voucher_objects_size_limit,
+        voucher_objs_number_limit,
+    )
+    .unwrap();
+}
+
+fn create_storage_bucket<T>(
+    lead_account_id: T::AccountId,
+    accepting_bags: bool,
+    bucket_objs_size_limit: u64,
+    bucket_objs_number_limit: u64,
+) where
+    T: Config + storage::Config,
+{
+    // Set storage bucket in the dynamic bag creation policy to zero.
+    let storage_wg_leader_signed = RawOrigin::Signed(lead_account_id);
+    Storage::<T>::create_storage_bucket(
+        T::Origin::from(storage_wg_leader_signed),
+        None,
+        accepting_bags,
+        bucket_objs_size_limit,
+        bucket_objs_number_limit,
+    )
+    .unwrap();
+}
+
+fn create_distribution_buckets<T>(
+    lead_account_id: T::AccountId,
+    distribution_bucket_family_id: T::DistributionBucketFamilyId,
+    bucket_number: u32,
+) -> BTreeSet<DistributionBucketId<T>>
+where
+    T: Config + storage::Config,
+{
+    let storage_wg_leader_signed = RawOrigin::Signed(lead_account_id);
+    (0..bucket_number)
+        .map(|_| {
+            let distribution_bucket_index =
+                Storage::<T>::distribution_bucket_family_by_id(distribution_bucket_family_id)
+                    .next_distribution_bucket_index;
+
+            Storage::<T>::create_distribution_bucket(
+                T::Origin::from(storage_wg_leader_signed.clone()),
+                distribution_bucket_family_id,
+                true,
+            )
+            .unwrap();
+
+            DistributionBucketId::<T> {
+                distribution_bucket_family_id,
+                distribution_bucket_index,
+            }
+        })
+        .collect::<BTreeSet<_>>()
+}
+
+fn create_distribution_bucket_with_family<T>(
+    lead_account_id: T::AccountId,
+    bucket_number: u32,
+) -> (
+    T::DistributionBucketFamilyId,
+    BTreeSet<DistributionBucketId<T>>,
+)
+where
+    T: Config + storage::Config,
+{
+    let storage_wg_leader_signed = RawOrigin::Signed(lead_account_id.clone());
+
+    let db_family_id = Storage::<T>::next_distribution_bucket_family_id();
+
+    Storage::<T>::create_distribution_bucket_family(T::Origin::from(storage_wg_leader_signed))
+        .unwrap();
+
+    (
+        db_family_id,
+        create_distribution_buckets::<T>(lead_account_id, db_family_id, bucket_number),
+    )
+}
+
+pub fn create_data_object_candidates_helper(
+    starting_ipfs_index: u8,
+    number: u64,
+    size: u64,
+) -> Vec<DataObjectCreationParameters> {
+    let range = (starting_ipfs_index as u64)..((starting_ipfs_index as u64) + number);
+
+    range
+        .into_iter()
+        .map(|_| DataObjectCreationParameters {
+            size,
+            ipfs_content_id: vec![1u8],
```

Comment:
This shouldn't be possible. `ipfs_content_id` of any length other than 46 bytes should not be accepted by the runtime. If this is not the case, it means that this important check is missing. This is bad, because it means `ipfs_content_id` is unbound and can be of any arbitrary length.

I created an issue to address that: https://github.com/Joystream/joystream/issues/4022